controller/instance_manager_controller.go

```diff
@@ -324,6 +327,10 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		if im.Status.CurrentState != types.InstanceManagerStateError {
 			im.Status.CurrentState = types.InstanceManagerStateStopped
 		}
+	} else if pod.DeletionTimestamp != nil {
+		if im.Status.CurrentState != types.InstanceManagerStateError {
+			im.Status.CurrentState = types.InstanceManagerStateError
+		}
 	} else {
 		// TODO: Will remove this reference kind correcting after all Longhorn components having used the new kinds
```

Comment:
Are these TODO's still relevant?

Reply:
This TODO is not related to this issue. We can create a separate PR for that.